Update README

sergiught opened the pull request:
<!--
❗ For general support or usage questions, use the Auth0 Community forums or raise a support ticket.

By submitting a pull request to this repository, you agree to the terms within the Auth0 Code of Conduct: https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md.
-->

### 🔧 Changes

General updates on the README. 
<!--
Describe both what is changing and why this is important. Include:

- Types and methods added, deleted, deprecated, or changed
- A summary of usage if this is a new feature or a change to a public API
-->

### 📚 References

<!--
Add relevant links supporting this change, such as:

- GitHub issue/PR number addressed or fixed
- Auth0 Community post
- StackOverflow answer
- Related pull requests/issues from other repositories

If there are no references, simply delete this section.
-->

### 🔬 Testing

<!--
Describe how this can be tested by reviewers. Be specific about anything not tested and why. Include any manual steps for testing end-to-end, or for testing functionality not covered by unit tests.
-->

### 📝 Checklist

- [ ] All new/changed/fixed functionality is covered by tests (or N/A)
- [x] I have added documentation for all new/changed functionality (or N/A)

<!--
❗ All the above items are required. Pull requests with an incomplete or missing checklist will be closed.
-->

willvedd commented on README.md:
```diff
@@ -54,77 +51,43 @@ provider "auth0" {}
+<div align="center">
 
-[Auth0](https://auth0.com/)
+<img alt="Auth0_black" src="https://user-images.githubusercontent.com/28300158/183676042-b9d92893-8fff-408f-9a36-63e77b14be30.png#gh-light-mode-only"  width="20%" height="20%">
```
Alt text here should vividly describe the content of the images.

```suggestion
<img alt="Auth0 logo and wordmark in black on transparent background" src="https://user-images.githubusercontent.com/28300158/183676042-b9d92893-8fff-408f-9a36-63e77b14be30.png#gh-light-mode-only"  width="20%" height="20%">
```

willvedd commented on README.md:
```diff
@@ -54,77 +51,43 @@ provider "auth0" {}
+<img alt="Auth0_black" src="https://user-images.githubusercontent.com/28300158/183676042-b9d92893-8fff-408f-9a36-63e77b14be30.png#gh-light-mode-only"  width="20%" height="20%">
 
-[[table of contents]](#table-of-contents)
+<img alt="Auth0_white" src="https://user-images.githubusercontent.com/28300158/183676141-bea463f9-af82-40ce-b18c-3a1030183d58.png#gh-dark-mode-only"  width="20%" height="20%">
```
```suggestion
<img alt="Auth0 logo and wordmark in white on transparent background" src="https://user-images.githubusercontent.com/28300158/183676141-bea463f9-af82-40ce-b18c-3a1030183d58.png#gh-dark-mode-only"  width="20%" height="20%">
```

willvedd commented on README.md:
```diff
@@ -54,77 +51,43 @@ provider "auth0" {}
+## 🙇 Support & Feedback
 
-## What is Auth0?
+### Raise an issue
```
```suggestion
### Raise an Issue
```

willvedd reviewed (commented):
Several small suggestions, but overall I think it's a nice refresh!

kevlil83 commented:
README is looking good. Can we update the description to be a little more.. descriptive? For example, "The official Terraform provider for managing Auth0 management API resources." Doesn't have to be exactly that but something that expands the description a little bit. Most will prob know what a Terraform provider is but others my stumble on it and it could be a learning opportunity.

kevlil83 commented:
Both of the links in the documentation section link to the same page. Not sure if that is a typo.

kevlil83 commented:
Small thing but I would add a little bit of white space between the logo and text on the bottom.

sergiught commented on README.md:
```diff
@@ -1,48 +1,45 @@
+### Requirements
 
--------------------------------------
+- [Terraform](https://www.terraform.io/downloads)
```
Not really, the provider works with any (v0 or v1).

sergiught commented on README.md:
```diff
@@ -54,77 +51,43 @@ provider "auth0" {}
-## Author
+---
+
+<div align="center">
```
This is following the SDK GitHub README and Docs Redesign format.

sergiught commented:
Thank you both for the review @kevlil83, @willvedd. I addressed all the comments. 

@kevlil83, I changed 1 link to point now to the github repo quickstart, as the docs haven't been published to the terraform registry yet (that happens only once we make a release). Afterwards I'll update the link to point to that quickstart.

willvedd reviewed (approved):
👍
